File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int mini = INT_MAX;
        int maxPro = 0;
        for(int i=0;i<n;i++){
            if(prices[i]<mini){
                mini = prices[i];
            }
            int pro = prices[i] - mini;
            if(pro>maxPro){
                maxPro = pro;
            }
        }
        return maxPro;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        //keeping minimum on left side of array.
        int maxi=0;
        int mini = Integer.MAX_VALUE;

        for(int i: prices){
            mini = Math.min(mini,i);
            int pro = i-mini;
            maxi = Math.max(maxi,pro);
        }
        return maxi;
    }
}